scona/visualisations.py

```diff
@@ -0,0 +1,347 @@
+    # so that there is one column for the degrees data, one for rich club values
+    # required for seaborn plotting with error bars
+
+    # create array to store the degrees
+    rand_degree = []
+
+    # create array to store a rich_club values according to the degree
+    rc_rand = []
+
+    # append each column in rand_df to a list
+    for i in range(len(rand_df.columns)):
+        rand_degree = np.append(rand_degree, rand_df.index.values)
+        rc_rand = np.append(rc_rand, rand_df.iloc[:, i])
+
+    new_rand_df = pd.DataFrame({'Degree': rand_degree, 'Rich Club': rc_rand})
+
+    # create a figure
+    fig, ax = plt.subplots(figsize=(10, 6))
+
+    # set the default colors of plotted values if not provided
+    if color is None:
+        color = ["#00C9FF", "grey"]
+    elif len(color) == 1:              # in case only to plot only real values
+        color.append("grey")
+
+    # if the given color not a list of size 2 - show warning, use default colors
+    if not isinstance(color, list) and len(color) != 2:
+        warnings.warn("Please, provide a *color* parameter as a "
+                      "python list object, e.g. [\"green\", \"pink\"]. "
+                      "Right now the default colors will be used")
+        color = ["#00C9FF", "grey"]
+
+    # plot the random rich club values of random graphs
+    ax = sns.lineplot(x="Degree", y="Rich Club", data=new_rand_df,
+                      err_style="band", ci=95, color=color[1],
+                      label="Random network", zorder=2)
+
+    # plot the rich club values of real Graph
+    ax = sns.lineplot(x=degree, y=rc_real, label="Real Network", zorder=1,
+                      color=color[0])
+
+    # set the max values of x & y - axis if not given
+    if x_max is None:
+        x_max = max(degree)
+
+    if y_max is None:
+        y_max = max(rc_real) + 0.1   # let y-axis be longer -> looks better
+
+    # set the x and y axis limits
+    ax.set_xlim((0, x_max))
+    ax.set_ylim((0, y_max))
+
+    # set the number of bins to 4
+    ax.locator_params(nbins=4)
+
+    # set the x and y axis labels
+    ax.set_xlabel("Degree")
+    ax.set_ylabel("Rich Club")
+
+    # create a legend if show_legend = True, otherwise - remove
+    if show_legend:
+        ax.legend(fontsize="x-small")
+    else:
+        ax.legend_.remove()
+
+    # remove the top and right spines from plot
+    sns.despine()
+
+    # adjust subplot params so that the subplot fits in to the figure area
+    plt.tight_layout()
+
+    # display the figure
+    plt.show()
+
+    # save the figure if the location-to-save is provided
+    if figure_name:
+        # use the helper-function from module helpers to save the figure
+        save_fig(fig, figure_name)
+        # close the file after saving to a file
+        plt.close(fig)
+
+
+def plot_network_measures(brain_bundle, real_network, figure_name=None,
+                          color=None, ci=95, show_legend=True):
+    """
```

Comment:
I'm going to ask you to replace `real_network` with something else throughout. 
My problem with "real" is that it's a fairly dubious way to describe a heavily simplified mathematical model. I think it's being used here in place of a much longer-to-state notion such as _"the proper network, the one you got from the mri data_" or _"the network you were interested in in the first place, before you made all those random ones"_

 I've got a couple of suggestions, I don't love any of them, but I prefer them all to _real_
* `main_network`  or simply `network`- this doesn't say very much but that's okay
* `distinguished_network`, `emphasized_network`, `featured_network` or `highlighted_network` - these suggest what the plotting function is going to do with them, which is highlight them over the other networks
* `initial_network` or `original_network` - this tells us we're talking about the network from which all the other networks were created.

Reply:
I like your suggestion to rename "real_network" to "original_network"

Reply:
I've put in some changes in https://github.com/wingedRuslan/scona/pull/3 to catch this in a few other places.